fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/service/LoanScheduleAssembler.java

```diff
@@ -1446,4 +1464,21 @@ public Pair<Loan, Map<String, Object>> assembleLoanApproval(AppUser currentUser,
 
         return Pair.of(loan, actualChanges);
     }
+
+    private boolean validateDownPaymentAttribute(final boolean isDownPaymentEnabledInLoanProduct, final JsonElement element) {
+        boolean isDownPaymentEnabled = isDownPaymentEnabledInLoanProduct;
+        if (!isDownPaymentEnabledInLoanProduct) {
+            if (this.fromApiJsonHelper.parameterExists(LoanProductConstants.ENABLE_DOWN_PAYMENT, element)) {
```

Comment:
This is wrong. We should only throw exception if the the down payment is disabled on the loan product (which is correctly checked) and when user tries to set down payment to be enabled during loan submission (this is the wrongly checked part)! If user provide enableDownPayment=false then nothing should happen: No violation!

Reply:
Done! Logic updated